department-of-veterans-affairs/caseflow
# Mpi csv end to end

Resolves issues with the end to end workflow with the MPI veteran extract download

### Description
Fully connected frontend to backend for MPI veteran_extract work, users will now be able to extract and download a file of all veterans needing MPI updates since the last time the same event was completed.

### Acceptance Criteria
- [ ] Clicking the generate button retrieves all the veterans that were added since the last SystemAdminEvent in a csv file.
- [ ] If there are no previous SystemAdminEvents, all veterans will be returned and downloaded once confirm is clicked.
- [ ] If no veterans are returned, the appropriate banner is displayed.
- [ ] If there is an error, a banner is displayed.

### Testing Plan
1. Go to caseflow user select and select system team admin
2. Ensure the feature toggle "sys_admin_page" is enabled
3. Go to misc. and click admin
4. Once in the admin page you should see System Admin Ui and a button generate
5. Click generate to retrieve all the veterans that were added since the last SystemAdminEvent in a csv file.
6. A banner or modal will pop up depending on the response of the veteran_extract
7. if Modal, click confirm to download the csv file


### User Facing Changes
**Admin Page**
![image](https://user-images.githubusercontent.com/108481161/201977487-2ec091a9-b3d6-46e9-8a08-2faf9bae9904.png)

**If no veterans are returned, the appropriate banner is displayed.**
![image](https://user-images.githubusercontent.com/108481161/201977614-6bfe0924-a454-4838-b7cd-fcd1ecc141a6.png)

**If there is an error, a banner is displayed.**
![image](https://user-images.githubusercontent.com/108481161/201978061-48dc4603-e1bd-4459-9b3d-5af8ce5ef131.png)

**Clicking the generate button retrieves all the veterans that were added since the last SystemAdminEvent in a csv file and downloaded once confirm is clicked**
![image](https://user-images.githubusercontent.com/108481161/201978313-b4bd26f1-685b-4cd7-9c73-10cccfe41393.png)
![image](https://user-images.githubusercontent.com/108481161/201978421-6a18a430-a85d-44ca-9069-0aa4abb3820c.png)
